fee-hook/src/FeeHook.sol

```diff
@@ -243,65 +139,38 @@ contract FeeHook is BaseHook, OApp, FunctionsClient{
-    * @return requestId The ID of the request
-    */
-    function withdrawFees() external returns (bytes32 requestId) {
-        if (block.chainid != destinationChainId) {
-            revert("Withdraw: not on destination chain");
+    function withdrawFees() external returns (uint256) {
+        address withdrawAddress = msg.sender;
+        uint256 totalSupply = IERC20(vMooneyAddress).totalSupply();
+        uint256 userBalance = IERC20(vMooneyAddress).balanceOf(msg.sender);
+        require(totalSupply > 0, "Total supply is zero");
+        require(userBalance > 0, "User balance is zero");
+        uint256 ts = block.timestamp;
+        if (lastSupplyUpdate == 0) {
+            lastSupplyUpdate = contractDeploymentTime;
         }
-        FunctionsRequest.Request memory req;
-        req.initializeRequestForInlineJavaScript(source); // Initialize the request with JS code
-        string memory addressString = uint256(uint160(msg.sender)).toHexString(20);
-        string[] memory args = new string[](1);
-        args[0] = addressString; // Set the address argument
-        req.setArgs(args); // Set the arguments for the request
-
-        // Send the request and store the request ID
-        requestId = _sendRequest(
-            req.encodeCBOR(),
-            subscriptionId,
-            gasLimit,
-            donID
-        );
-        requestIdToSender[requestId] = msg.sender;
-
-        return requestId;
-    }
+        supplyTimeWeighted += totalSupply * (ts - lastSupplyUpdate);
+        lastSupplyUpdate = ts;
 
-    /**
-     * @notice Callback function for fulfilling a request to withdrawFees
-     * @param requestId The ID of the request to fulfill
-     * @param response The HTTP response data
-     * @param err Any errors from the Functions request
-     */
-    function fulfillRequest(
-        bytes32 requestId,
-        bytes memory response,
-        bytes memory err
-    ) internal override {
-        if (err.length > 0) {
-            revert(string(err));  // bubble up the actual error
+        if (lastBalanceUpdate[withdrawAddress] == 0) {
+            lastBalanceUpdate[withdrawAddress] = contractDeploymentTime;
         }
-        // Update the contract's state variables with the response and any errors
-        (uint256 totalSupply, uint256 userBalance) = abi.decode(response, (uint256, uint256));
-        require(totalSupply > 0, "Total supply is zero");
-        require(userBalance > 0, "User balance is zero");
-        uint256 userProportion = userBalance * 1e18 / totalSupply; // Multiply by 1e18 to preserve precision
-        uint256 allocated = (userProportion * totalReceived) / 1e18; // Divide by 1e18 to normalize
-        address withdrawAddress = requestIdToSender[requestId];
-        require(withdrawAddress != address(0), "Unknown requestId");
+        balanceTimeWeighted[withdrawAddress] += userBalance * (ts - lastBalanceUpdate[withdrawAddress]);
```

Comment:
So this assumes their userBalance was the same throughout that time period right? In reality it's decreasing linearly as a function of the lock time and lock amount. We could potentially do that math here but maybe it's overkill?